src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -537,32 +537,29 @@ private class WaitDialogProgress : RestoreOperationProgressUI
                 JoinableTaskFactory jtf,
                 CancellationToken token)
             {
-                return await jtf.RunAsync(async () =>
-                {
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                    var waitDialogFactory = serviceProvider.GetService<
-                        SVsThreadedWaitDialogFactory, IVsThreadedWaitDialogFactory>();
-
-                    var session = waitDialogFactory.StartWaitDialog(
-                        waitCaption: Resources.DialogTitle,
-                        initialProgress: new ThreadedWaitDialogProgressData(
-                            Resources.RestoringPackages,
-                            progressText: string.Empty,
-                            statusBarText: string.Empty,
-                            isCancelable: true,
-                            currentStep: 0,
-                            totalSteps: 0));
-
-                    return new WaitDialogProgress(session, jtf);
-                });
+                await jtf.SwitchToMainThreadAsync();
+
+                var waitDialogFactory = serviceProvider.GetService<
+                    SVsThreadedWaitDialogFactory, IVsThreadedWaitDialogFactory>();
+
+                var session = waitDialogFactory.StartWaitDialog(
```

Comment:
(Looking at just this method) I'm confused at why an *async* method would be initializing the threaded wait dialog. That service is for methods that must synchronously block the UI thread. And in fact for that purpose we have a convenient helper extension method (an overload of JTF.Run) that will display the TWD for you. 
So if you actually are doing something asynchronously, consider removing use of TWD to match guidelines.
Or if this method's caller uses JTF.Run, I would suggest you let that method call the Run overload that displays TWD.
Either way, I think this method goes away.